tests/test_provision.py

```diff
@@ -229,3 +229,11 @@ def test_provision_conf_file(tox_project: ToxProjectCreator, tmp_path: Path, rel
     conf_path = str(Path(project.path.name) / "tox.ini") if relative_path else str(project.path / "tox.ini")
     result = project.run("c", "--conf", conf_path, "-e", "py", from_cwd=tmp_path)
     result.assert_success()
+
+
+@pytest.mark.parametrize("subcommand", ["r", "p", "de", "l", "d", "c", "q", "e", "le"])
+def test_provision_default_arguments_exists(tox_project: ToxProjectCreator, subcommand: str) -> None:
+    ini = "[tox]\nrequires = tox<4"
```

Comment:
This test is heavier that it needs to be, can we make it faster and simpler?

Reply:
```diff
 @pytest.mark.parametrize("subcommand", ["r", "p", "de", "l", "d", "c", "q", "e", "le"])
 def test_provision_default_arguments_exists(tox_project: ToxProjectCreator, subcommand: str) -> None:
-    ini = "[tox]\nrequires = tox<4"
-    outcome = tox_project({"tox.ini": ini}).run(subcommand)
+    ini = r"""
+    [tox]
+    requires =
+        tox<4.14
+    [testenv]
+    package = skip
+    """
+    project = tox_project({"tox.ini": ini})
+    project.patch_execute(lambda r: 0 if "install" in r.run_id else None)
+    outcome = project.run(subcommand)
     for argument in ["result_json", "hash_seed", "discover", "list_dependencies"]:
         assert hasattr(outcome.state.conf.options, argument)
```

This makes the tests run in 360ms instead of 4 seconds.